R/utils.R

```diff
@@ -132,8 +132,12 @@ eval_with_cols = function(orig_call, all_cols) {
   })
   if (!is.primitive(fun)) {
     named_call = match.call(fun, orig_call)
-    if ("cols" %in% names(formals(fun)) && !"cols" %in% names(named_call)) {
-      named_call[["cols"]] = all_cols
+    if ("cols" %in% names(formals(fun))) {
+      if ("cols" %in% names(named_call)) {
```

Comment:
Unnecessary nesting here?

```suggestion
    if ("cols" %in% names(formals(fun)) && "cols" %in% names(named_call)) {
```

Reply:
NVM, I'm wrong, the nesting is necessary because of the `else`.